numba/cuda/cudadrv/nvvm.py

```diff
@@ -650,10 +714,51 @@ def patch_ptx_debug_pubnames(ptx):
 
 re_parenthesized_list = re.compile(r"\((.*)\)")
 
+re_spflags = re.compile(r"spFlags: (.*),")
+
+spflagmap = {
+    'DISPFlagDefinition': 'isDefinition',
+    'DISPFlagOptimized': 'isOptimized',
+}
+
 
-def llvm39_to_34_ir(ir):
+def llvm100_to_70_ir(ir):
     """
-    Convert LLVM 3.9 IR for LLVM 3.4.
+    Convert LLVM 10.0 IR for LLVM 7.0.
+    """
+    buf = []
+    for line in ir.splitlines():
+        if line.startswith('attributes #'):
+            # Remove function attributes unsupported by LLVM 7.0
+            m = re_attributes_def.match(line)
+            attrs = m.group(1).split()
+            attrs = ' '.join(a for a in attrs if a != 'willreturn')
+            line = line.replace(m.group(1), attrs)
+
+        if '!DISubprogram' in line:
+            # Replace the DISPFlags (LLVM 10.0) with main subprogram DIFlags
+            # (LLVM 7.0). Example:
+            #
+            #     spflags: DISPFlagDefinition | DISPFlagOptimized
+            #
+            # becomes:
+            #
+            #     isDefinition: true, isOptimized: true
+            m = re_spflags.search(line)
+            flags = m.group(1).split(' | ')
+            new_flags = ", ".join([ '%s: true' % spflagmap[f] for f in flags ])
+            start_of_line = line[:m.span()[0]]
+            end_of_line = line[m.span()[1] - 1:]
+            line = start_of_line + new_flags + end_of_line
```

Comment:
Does this cover all cases of `DISPFlags` that might appear?

Reply:
These are all that I've encountered during testing. Since there doesn't appear to be any explicit enumeration / explanation of these flags in the documentation, and it's not necessarily clear what the mapping for all the ones I haven't seen should be, I'm inclined to leave this as-is and add support for others if/when they are encountered (e.g. when changing the LLVM version used by llvmlite, or when enhancing debugging support in Numba / llvmlite).

Reply:
ok, this seems reasonable, will deal with on a case by case basis as they occur as we can't easily preempt it.